### PR TITLE
Task 2: Progress Button and Error Handler

### PR DESCRIPTION
**Pull request by Ananth-Desai**

### CODE REVIEW

File: Flickr/Sources/FlickrAppVCs/TabsVC.swift [hunk 1]
```diff
@@ -43,5 +43,5 @@ class TabsVC: UITabBarController {
 
 private let tabbarBackgroundColor = UIColor(red: 0.976, green: 0.976, blue: 0.976, alpha: 0.94)
 private let tabbarTintColor = UIColor(red: 0.952, green: 0.219, blue: 0.474, alpha: 1.0)
-private let homeVcTitle = "Search"
-private let favoritesVcTitle = "Favorites"
+private let homeVcTitle = NSLocalizedString("search", comment: "")
```
Comment (Purushothaman22) [inline Flickr/Sources/FlickrAppVCs/TabsVC.swift:46]: @Ananth-Desai  I assume R.swift is there in project. So here you can just use, `R.string.localisable.search`. Please check and update.
```suggestion
private let homeVcTitle = NSLocalizedString("search", comment: "")
```

**Review by Purushothaman22 — CHANGES_REQUESTED** [review]

File: Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift [hunk 2]
```diff
@@ -6,9 +6,14 @@
 //
 
 import Foundation
+import LeoUI
 import UIKit
 
 class HomeScreenVC: UIViewController {
+    private var progressButton: ProgressButton?
+    private var searchButton: UIButton?
+    private var searchString: UITextField?
```
Comment (Purushothaman22) [inline Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift]: Shouldn't these views be weak?

File: Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift [hunk 3]
```diff
@@ -24,51 +29,106 @@ class HomeScreenVC: UIViewController {
...
+        view.addSubview(textField)
         return [
-            label.centerXAnchor.constraint(equalTo: view.centerXAnchor),
-            label.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: -150),
-            label.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 30),
-            label.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -30)
+            textField.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            textField.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: -150),
+            textField.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 30),
+            textField.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -30)
```
Comment (Purushothaman22) [inline Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift]: Declare constant at EOF so it is easy to make changes later

File: Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift [hunk 4]
```diff
@@ -24,51 +29,106 @@ class HomeScreenVC: UIViewController {
         }
         let searchButtonConstraints = setupSearchButton()
         NSLayoutConstraint.activate(searchButtonConstraints)
+
+        let progressButtonConstraints = setupProgressButton()
+        NSLayoutConstraint.activate(progressButtonConstraints)
     }
 
     @available(iOS 13.0, *)
```
Comment (Purushothaman22) [inline Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift]: Please use UISearchBar

File: Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift [hunk 5]
```diff
@@ -24,51 +29,106 @@ class HomeScreenVC: UIViewController {
...
         view.addSubview(button)
 
         return [
             button.centerXAnchor.constraint(equalTo: view.centerXAnchor),
             button.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: -100)
         ]
     }
+
+    private func setupProgressButton() -> [NSLayoutConstraint] {
+        let progress = ProgressButton()
```
Comment (Purushothaman22) [inline Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift]:
```suggestion
        let progressButton = ProgressButton()
```

File: Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift [hunk 6]
```diff
@@ -24,51 +29,106 @@ class HomeScreenVC: UIViewController {
...
+            progress.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            progress.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: -100),
+            progress.leadingAnchor.constraint(equalTo: view.centerXAnchor, constant: -50),
+            progress.trailingAnchor.constraint(equalTo: view.centerXAnchor, constant: 50),
+            progress.topAnchor.constraint(equalTo: searchString.bottomAnchor, constant: 10)
+        ]
+    }
+
+    @objc private func setButtonBackground() {
+        searchButton?.isEnabled = (searchString?.text!.count)! > 0 ? true : false
```
Comment (Purushothaman22) [inline Flickr/Sources/FlickrAppVCs/HomeScreenVC.swift]: Please don't use force unwrap anywhere else

**Review by Purushothaman22 — CHANGES_REQUESTED** [review]

File: Flickr/Sources/FlickrAppVCs/TabsVC.swift (hunk 1 above)
Comment (Ananth-Desai) [reply Flickr/Sources/FlickrAppVCs/TabsVC.swift:46]: `R.string.localisable.search` gives an error saying `Type 'R' has no member 'string'`

**Review by Purushothaman22 — APPROVED** [review]

**Pull request merged**